polyglot/piranha/src/lib.rs

```diff
@@ -93,11 +96,30 @@ pub fn execute_piranha(
   }
 
   // flag_cleaner.relevant_files
-  flag_cleaner
+  let summaries = flag_cleaner
     .get_updated_files()
     .iter()
     .map(PiranhaOutputSummary::new)
-    .collect_vec()
+    .collect_vec();
+  log_the_summary(&summaries);
+  return summaries;
+}
+
+fn log_the_summary(summaries: &Vec<PiranhaOutputSummary>) {
+    let mut total_number_of_matches : usize = summaries.iter().map(|x|x.matches().len()).sum();
+    let mut total_number_of_rewrites : usize = summaries.iter().map(|x|x.rewrites().len()).sum();
+    for summary in summaries {
+      let number_of_rewrites =&summary.rewrites().len();
+      let number_of_matches = &summary.matches().len();
+      info!("File : {:?}", &summary.path());
+      info!("{}", format!("# Rewrites : {number_of_rewrites}"));
+      info!("{}", format!("# Matches : {number_of_matches}"));
+      total_number_of_rewrites += number_of_rewrites;
```

Comment:
Aren't this and the next line double-counting if we already computed `total_number_of_rewrites` as `summaries.iter().map(|x|x.rewrites().len()).sum()` above? I think this ends up printing 2x the total number of rewrites...

Reply:
Good catch.